### DISCUSSION TITLE
How are fallback colors calculated for out-of-gamut tints and shades, and where can I find this functionality in the code?

### DISCUSSION THREAD

**Discussion by eduard-buta** [evilmartians/oklch-picker#164] [Q&A] [answered]
Hi there. I'm trying to write a script for Adobe Illustrator, and I want to use the OKLCH color space. I've run into some out-of-gamut problems when trying to generate some specific hues, and I wanted to ask for your help in pinpointing the part of the source code that handles these fallbacks just like in the web tool. Thank you very much!

**Comment by ai** [comment, marked as answer]
We use [`toGamut('rgb', 'oklch')`](https://github.com/evilmartians/oklch-picker/blob/main/lib/colors.ts#L109C20-L109C33) from [Culori](https://culorijs.org/api/#toGamut)
